### ISSUE TITLE
Unable to find C++ package in editable mode

### ISSUE THREAD

**Issue by JeremyBYU** [scikit-build/scikit-build-core#304]
Thanks for this great tool. I am building a pybind11 c++ package using `scikit-build-core`.  I am using the template as found [here](https://github.com/pybind/scikit_build_example). Note: I am pretty sure the issue I am facing has not much to do with pybind11.

Everything works and the package is built successfully when I install with `pip install .`. However, when I build in _editable_ mode, `pip install -e .`, the **compiled** binary module fails to load. Here is the error:

```
❯ python
Python 3.10.11 (main, Apr 20 2023, 13:58:42) [Clang 14.0.6 ] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import pymultiastar
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/jcastagno/Documents/Research/pymultiastar/src/pymultiastar/__init__.py", line 1, in <module>
    from ._core import __doc__, __version__, PyMultiAStar
ModuleNotFoundError: No module named 'pymultiastar._core'
```

This file (`pymultiastar._core.cpython-310-darwin.so` ) is indeed created and can be found in my `site_packages/pymulitastar` folder in my virtual environment. I think I have identified *why* it can not find `_core`.  It has to do with me *manually* adding  a _core.py**i** that provided **typing** to this compiled module. 

 Looking in the site packages folder I can see another file called `_pymultiastar_editable.py`. It seems this file is what allows you to have a link to your development python files. I noticed something near the bottom:

```python
....
install({'pymultiastar._core': '/Users/jcastagno/Documents/Research/pymultiastar/src/pymultiastar/_core.pyi', ...
{'pymultiastar._core': 'pymultiastar/_core.cpython-310-darwin.so'}
```

It seems it uses this dictionary to help specify **where** the files of the package will be located. Notice that it has **two** `pymultiastar._core`, one for my local editable _core.pyi python typing and the other for the compiled module. 

If I remove this _first_ dictionary key (.pyi), then everything works, even in editable mode. So...is there a way where we can tell scikit-build-core to ignore the .pyi files in editable mode?

Thanks!

**Comment by henryiii** [comment]
Should be fixed in 0.3.3!

**Comment by JeremyBYU** [comment]
Thank you so much! I appreciate your prompt help on this!
